### DISCUSSION TITLE
How to utilize useContext in expo router?

### DISCUSSION THREAD

**Discussion by gibumkang** [expo/router#641] [Q&A] [answered]
From the docs, it sounds like _layout.tsx in the root of /app is the root file. However, it doesn't allow for hooks to be added on that file. I have tried to wrap my Provider around index.tsx instead but that doesn't seem to work. What is the correct way to utilize useContext to pass state across different pages?

**Comment by marklawlor** [comment, marked as answer]
You can use hooks within `_layout` files - they are just React components. If you want a global context, then you should add the Provider to the top level `_layout`

**Comment by gibumkang** [comment]
I just realized my root `_layout` file has the `useFonts` utility with the following:
```
if(!fontsLoaded){
   return null
}
```
This codeblock has to be on the bottom, otherwise it throws an error. I've moved my context above it and now it's working. Thank you!
